__tests__/profile/change-profile.test.tsx

```diff
@@ -0,0 +1,100 @@
+// import React from 'react'
+// import { render, fireEvent, waitFor } from '@testing-library/react'
+// import ChangeProfile from '@/app/profile/change-profile/page'
+// import {
+//   useUpdateProfileMutation,
+//   useGetProfileQuery,
+// } from '@/redux/api/profileApi'
+// import { useDispatch } from 'react-redux'
+// import { store } from '@/redux/store'
+// import { setProfile } from '@/redux/features/profileSlice'
+// import { Provider } from 'react-redux'
+// import { useRouter } from 'next/navigation'
+
+// jest.mock('@/redux/api/profileApi', () => ({
+//   useUpdateProfileMutation: jest.fn(),
+//   useGetProfileQuery: () => ({
+//     data: {
+//       profile: {
+//         bio: 'hi',
+//       },
+//     },
+//   }),
+// }))
+
+// jest.mock('react-redux', () => ({
+//   ...jest.requireActual('react-redux'),
+//   useDispatch: jest.fn(),
+// }))
+
+// jest.mock('next/navigation', () => ({
+//   useRouter: jest.fn(),
+// }))
```

Comment:
Rather than mocking useRouter, I suggest you to use <Link> instead on your page as it does not need mocking when doing tests.